Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -92,6 +96,119 @@ namespace gdjs {
           .replace(/[^\w|-]/g, '')
           .slice(0, 30);
       };
+
+      const checkLeaderboardAvailability = async function (
+        url: string
+      ): Promise<boolean> {
+        try {
+          const response = await fetch(url, {
+            method: 'GET',
+            headers: { 'Content-Type': 'application/json' },
+          });
+          if (!response.ok) {
+            logger.error(
+              `Error while fetching leaderboard view, server returned: ${response.status} ${response.statusText}`
+            );
+            return false;
+          }
+          return true;
+        } catch (err) {
+          logger.error(`Error while fetching leaderboard view: ${err}`);
+          return false;
+        }
+      };
+
+      const receiveMessage = function (
+        runtimeScene: gdjs.RuntimeScene,
+        event: MessageEvent
+      ) {
+        if (event.data === 'closeLeaderboardView') {
+          closeLeaderboardView(runtimeScene);
+        }
+      };
+
+      export const displayLeaderboard = async function (
+        runtimeScene: gdjs.RuntimeScene,
+        leaderboardId: string
+      ) {
+        const gameId = gdjs.projectData.properties.projectUuid;
+        const targetUrl = `https://liluo.io/games/${gameId}/leaderboard/${leaderboardId}`;
+        if (!(await checkLeaderboardAvailability(targetUrl))) {
+          logger.error('Leaderboard data could not be fetched, doing nothing');
+          return;
+        }
+
+        if (_leaderboardViewIframe) {
+          _leaderboardViewIframe.src = targetUrl;
+        } else {
+          const domElementContainer = runtimeScene
+            .getGame()
+            .getRenderer()
+            .getDomElementContainer();
+          if (!domElementContainer) {
+            logger.error(
+              "The div element covering the game couldn't be found, the leaderboard cannot be displayed."
+            );
+            return;
+          }
+          const iframe = document.createElement('iframe');
+
+          iframe.src = targetUrl;
```

Comment:
For the loading error handling:
- Before setting `src`, add some `addEventListener` to listen to `load` and `error`. 
- In case of error, log it and set  the `_leaderboardViewIframeErrored` to true, then close the leaderboard view (the iframe could not load, we don't want to soft lock the game on a blank screen).
- In the "load" event: Set it to false, for robustness/safety sake.